samcli/commands/pipeline/bootstrap/pipeline_oidc_provider.py

```diff
@@ -0,0 +1,61 @@
+"""
+Represents a pipeline OIDC provider
+"""
+from abc import abstractmethod
+from typing import List
+import click
+
+from samcli.lib.config.samconfig import SamConfig
+
+
+class PipelineOidcProvider:
+
+    PROVIDER_URL_PARAMETER = "oidc-provider-url"
+    CLIENT_ID_PARAMETER = "oidc-client-id"
+
+    def __init__(self, oidc_parameters: dict, oidc_parameter_names: List[str]) -> None:
+        self.oidc_parameters = oidc_parameters
+        self.oidc_parameter_names = [self.PROVIDER_URL_PARAMETER, self.CLIENT_ID_PARAMETER] + oidc_parameter_names
+
+    def verify_parameters(self) -> None:
```

Comment:
If this is needed for the object to considered correct, should this be in the constructor instead?